### DISCUSSION TITLE
Best way to dump a push notification for debugging?

### DISCUSSION THREAD

**Discussion by kelfink** [relayrides/pushy#981] [Q&A] [answered]
Is there a convenient way to dump a push notification rather than actually send it with sendNotification ?
I'd like to have a verification step for working with other teams and debugging so we can validate that a send to APNS is what the iOS app expects.  

E.g. I want to see "expiration-time: 0" in the output instead of indicating I set invalidationTime of null.

Thanks!

**Comment by jchambers** [comment, marked as answer]
The easiest thing to do here would probably be to [turn on HTTP/2 frame logging](https://pushy-apns.org/apidocs/latest/com/eatthepath/pushy/apns/ApnsClientBuilder.html#setFrameLogger(io.netty.handler.codec.http2.Http2FrameLogger)). If you coupled that with a [mock server](https://pushy-apns.org/apidocs/latest/com/eatthepath/pushy/apns/server/MockApnsServer.html), I think you'd get what you're looking for.

**Reply by kelfink** [reply]
sounds good, thanks.
